### PR TITLE
[New Rule] Unusual Network Connection via DllHost

### LINKED ISSUE #1231: [New Rule] Unusual Network Connection via DllHost
## Description
Identifies unusual instances of dllhost.exe making outbound network connections. This may indicate adversarial Command
and Control activity.

### Target indexes
`"winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"`

### Target Operating Systems
`windows`

### Query
```
sequence by host.id, process.entity_id with maxspan=1m
  [process where event.type in ("start", "process_started") and process.name : "dllhost.exe" and process.args_count == 1]
  [network where process.name : "dllhost.exe" and
   not cidrmatch(destination.ip, "10.0.0.0/8", "127.0.0.0/8", "169.254.0.0/16", "172.16.0.0/12", "192.0.0.0/24", 
    "192.0.0.0/29", "192.0.0.8/32", "192.0.0.9/32", "192.0.0.10/32", "192.0.0.170/32", "192.0.0.171/32", "192.0.2.0/24",
    "192.31.196.0/24", "192.52.193.0/24", "192.168.0.0/16", "192.88.99.0/24", "224.0.0.0/4", "100.64.0.0/10", 
    "192.175.48.0/24", "198.18.0.0/15", "198.51.100.0/24", "203.0.113.0/24", "240.0.0.0/4", "::1", "FE80::/10", 
    "FF00::/8")]  
```

### References
https://www.volexity.com/blog/2021/05/27/suspected-apt29-operation-launches-election-fraud-themed-phishing-campaigns/
https://www.microsoft.com/security/blog/2021/05/27/new-sophisticated-email-based-attack-from-nobelium/

### PR DESCRIPTION
**Pull request by bm11100**
## Issues
resolves #1231 

## Summary
Identifies unusual instances of dllhost.exe making outbound network connections. This may indicate adversarial Command and Control activity.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Review by dstepanic17 — APPROVED** [review]

**Review by Samirbous — APPROVED** [review]
👍

File: rules/windows/defense_evasion_unusual_network_connection_via_dllhost.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+references = [
+    "https://www.microsoft.com/security/blog/2021/05/27/new-sophisticated-email-based-attack-from-nobelium/",
+    "https://www.volexity.com/blog/2021/05/27/suspected-apt29-operation-launches-election-fraud-themed-phishing-campaigns/",
+    "https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml",
+]
+risk_score = 47
+rule_id = "c7894234-7814-44c2-92a9-f7d851ea246a"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+type = "eql"
```
Comment (brokensound77) [inline rules/windows/defense_evasion_unusual_network_connection_via_dllhost.toml:26]: `timestamp_override`?

Comment (bm11100) [reply rules/windows/defense_evasion_unusual_network_connection_via_dllhost.toml:27]: oh yes

**Review by brokensound77 — APPROVED** [review]
🚀

**Pull request merged**